File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float dollar;
    do
    {
        dollar = get_float("change owned: ");
    } while (dollar < 0);
    
   
   int cents = round(dollar * 100);
   int coins = 0;
   
   int denominations[] = {25, 10, 5, 1};
   int size = 4;
   for (int i = 0; i < size; i++)
   //bad design
   {
   coins += cents / denominations[i];
   cents %= denominations[i];
   }
   printf("%i\n", coins);

    
}

File: pset6/cash.py
from cs50 import get_float
#set coin counter to 0
coins = 0
#ask for input; stop asking only when input is valid (i.e. > 0)
while True:
    change_dollar = get_float("Change (dollars): ")
    if change_dollar > 0:
        break
#convert it into cents and round
change = round(int(change_dollar * 100))
#for every coin sorted, the counter +1
while change > 0:
    while change >= 25:
        coins += 1
        change -= 25
    while change >= 10:
        coins += 1
        change -= 10
    while change >= 5:
        coins += 1
        change -= 5
    while change >= 1:
        change -= 1
        coins += 1
print(coins)